File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (!root) return root;

        if (root->left and root->right){
            TreeNode* temp = root->left;
            root->left = root->right;
            root->right = temp;
        }
        else if (root->left){
            root->right = root->left;
            root->left = nullptr;
        }
        else{
            root->left = root->right;
            root->right = nullptr;
        }
        
        if (root->left) invertTree(root->left);
        if (root->right) invertTree(root->right);

        return root;
    }
};

File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if not root: return TreeNode(val)
        
        def helper(node):
            if not node:
                return 

            if val < node.val:
                if not node.left:
                    node.left = TreeNode(val)
                    return 
                
                return self.insertIntoBST(node.left, val)

            if val > node.val:
                if not node.right:
                    node.right = TreeNode(val)
                    return

                return self.insertIntoBST(node.right, val)

        helper(root)
        return root
